src/compiler.ts

```diff
@@ -6992,6 +6994,8 @@ export class Compiler extends DiagnosticEmitter {
       if (classInstances) {
         for (let _values = Map_values(classInstances), j = 0, l = _values.length; j < l; ++j) {
           let classInstance = _values[j];
+          // Chcek if the parent class is a subtype of instance's class
+          if (!classInstance.isAssignableTo(parentClassInstance)) continue;
```

Comment:
More an intuition right now, but could you elaborate why this is the right check?

Reply:
Well assuming `isAssingableTo` works as `isSubClassOf`, then consider:

```ts
class A<T> {
  foo(t: T): string { return "A";}
  }
  
class B extends A<i32>{
 foo(t: i32): string { return "B";}
}

class C extends A<string>{
 foo(t: string): string { return t; }
 }

let a: A<i32> = new B();
let a: A<string> = new C();
```

Currently given the way that virtual methods are collected, the instance of `A<i32>`'s `foo` method would have both `B` and `C` in the `classInstances` above. So this check is when generating the virtual table for `A<i32>`s `foo` method, only subclasses of `A<i32>` should be considered.

Reply:
@dcodeIO @MaxGraey

Reply:
If @dcodeIO confirms this PR as correct, we'll be merge it all soon with some PRs marked as "feat"

Reply:
I *think* this looks correct to me. I do not think that we'll have to make it a breaking 0.20, though :)